### PR TITLE
Issue #39

### PR DESCRIPTION
**Pull request by cygnus-x-1**
### Description of the Change

Added support for logging of views in Constellation.
Updated AbstractTopComponent to log view behaviour.

### Alternate Designs

Logging in Constellation is standardised through the ConstellationLogger class, so that was used in lieu of alternate designs.

### Why Should This Be In Core?

Because logging is important :)

### Benefits

Logging views allows tracking of which views are used, and how they are used in order to inform updates and improvements.

### Possible Drawbacks

Views not implementing AbstractTopComponent currently have no logging implemented.

### Verification Process

Refactored DefaultConstellationLogger to optionally log to standard out and then performed exploratory testing to ensure all expected log messages were being sent.

### Applicable Issues

N/A

### CODE REVIEW

File: CoreScripting/src/au/gov/asd/tac/constellation/views/scripting/utilities/ScriptingUtilities.java [hunk 1]
```diff
@@ -101,7 +100,6 @@ public SGraph copyGraph(final SGraph graph) {
     public void executePlugin(final SGraph graph, final String pluginName) {
         final Plugin plugin = PluginRegistry.get(pluginName);
         final PluginParameters parameters = new PluginParameters();
-        parameters.appendParameters(GlobalParameters.getParameters(null));
```
Comment (arcturus2) [inline CoreScripting/src/au/gov/asd/tac/constellation/views/scripting/utilities/ScriptingUtilities.java]: What was the reason behind removing this line?

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,9 @@
 * Fixed the GitHub url used by Help -> Submit a ticket.
 * Removed several unused dependencies, including JOGL, JTS, OpenCSV, Trove4j, JScience, and XML-APIs.
 * Updated several dependencies to the latest versions, including Geotools, Jetty, Apache Commons, Jackson, RSyntaxArea, Google Guava, Apache POI, EJML, Processing, Jython, and SwingX.
+* Updated ConstellationLogger new methods viewStarted, viewStopped and viewInfo to support logging of Views.
```
Comment (arcturus2) [inline CHANGELOG.md]: A minor comment but should this read "Updated ConstellationLogger *with* new methods..." ?

File: CorePluginFramework/src/au/gov/asd/tac/constellation/pluginframework/logging/DefaultConstellationLogger.java [hunk 3]
```diff
@@ -19,44 +19,79 @@
...
  *
  * @author sirius
+ * @author cygnus_x-1
  */
 @ServiceProvider(service = ConstellationLogger.class, position = Integer.MAX_VALUE)
 public class DefaultConstellationLogger implements ConstellationLogger {
 
+    private static final Logger LOGGER = Logger.getLogger(DefaultConstellationLogger.class.getName());
+
+    private static final boolean VERBOSE = false;
```
Comment (arcturus2) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/pluginframework/logging/DefaultConstellationLogger.java]: I'd avoid doing it this way. The better approach would be to set the log level to `Level.FINE` and you activate it by passing  `-J-Dau.gov.asd.tac.constellation.pluginframework.logging.DefaultConstellationLogger.Level=FINE` in `project.properties`. This means that you can enable/disable logging without recompiling.

File: CorePluginFramework/src/au/gov/asd/tac/constellation/pluginframework/logging/ConstellationLogger.java [hunk 4]
```diff
@@ -34,47 +36,71 @@ public static ConstellationLogger getDefault() {
     }
 
     /**
-     * Called to indicate that Constellation is starting.
+     * Called to indicate that Constellation was started.
      */
-    public void applicationStart();
+    public void applicationStarted();
```
Comment (arcturus2) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/pluginframework/logging/ConstellationLogger.java]: I like the new names @cygnus-x-1 👍.
This is however an API change so can you please update CHANGELOG.md with the methods that have been renamed please?

File: CoreScripting/src/au/gov/asd/tac/constellation/views/scripting/utilities/ScriptingUtilities.java (hunk 1 above)
Comment (cygnus-x-1) [reply CoreScripting/src/au/gov/asd/tac/constellation/views/scripting/utilities/ScriptingUtilities.java]: It was unnecessary to add all global parameters to plugins (plugins specify if they need global parameters themselves), and this removes a dependency on the Data Access View module (which makes things more modular).

File: CHANGELOG.md (hunk 2 above)
Comment (cygnus-x-1) [reply CHANGELOG.md]: Yep - fixed.

File: CorePluginFramework/src/au/gov/asd/tac/constellation/pluginframework/logging/DefaultConstellationLogger.java (hunk 3 above)
Comment (cygnus-x-1) [reply CorePluginFramework/src/au/gov/asd/tac/constellation/pluginframework/logging/DefaultConstellationLogger.java]: I was considering that - fixed.

File: CorePluginFramework/src/au/gov/asd/tac/constellation/pluginframework/logging/ConstellationLogger.java (hunk 4 above)
Comment (cygnus-x-1) [reply CorePluginFramework/src/au/gov/asd/tac/constellation/pluginframework/logging/ConstellationLogger.java]: No worries - fixed.

**Pull request merged**